<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>896</width>
    <height>701</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>896</width>
    <height>701</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>896</width>
    <height>701</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Serial PWM Controls for ATmega328P</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="graphGroup">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>871</width>
      <height>261</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="title">
     <string>PWM Graph</string>
    </property>
    <widget class="QCustomPlot" name="graphWidget" native="true">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>30</y>
       <width>851</width>
       <height>221</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>871</width>
      <height>161</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="TAB1">
     <attribute name="title">
      <string>Software PWM</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>851</width>
        <height>121</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="title">
       <string>PWM Controls</string>
      </property>
      <widget class="QPushButton" name="setButton">
       <property name="geometry">
        <rect>
         <x>760</x>
         <y>30</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Set</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="FrequencyGroup">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>241</width>
         <height>91</height>
        </rect>
       </property>
       <property name="title">
        <string>Frequency (Hz)</string>
       </property>
       <widget class="QDoubleSpinBox" name="FrequencySpinBox">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>20</y>
          <width>191</width>
          <height>31</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="minimum">
         <double>100.000000000000000</double>
        </property>
        <property name="maximum">
         <double>8000.000000000000000</double>
        </property>
        <property name="value">
         <double>300.000000000000000</double>
        </property>
       </widget>
       <widget class="QSlider" name="FrequencySlider">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>60</y>
          <width>191</width>
          <height>21</height>
         </rect>
        </property>
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>8000</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBelow</enum>
        </property>
        <property name="tickInterval">
         <number>1000</number>
        </property>
       </widget>
      </widget>
      <widget class="QGroupBox" name="DutyCycleGroup">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>20</y>
         <width>241</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="title">
        <string>Duty Cycle (%)</string>
       </property>
       <widget class="QDoubleSpinBox" name="Dut">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>20</y>
          <width>201</width>
          <height>31</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="value">
         <double>50.000000000000000</double>
        </property>
       </widget>
       <widget class="QSlider" name="DutyCycleSlider">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>60</y>
          <width>191</width>
          <height>21</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBelow</enum>
        </property>
       </widget>
      </widget>
      <widget class="QCheckBox" name="PWMCheck">
       <property name="geometry">
        <rect>
         <x>540</x>
         <y>70</y>
         <width>301</width>
         <height>31</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Inverted PWM</string>
       </property>
      </widget>
      <widget class="QComboBox" name="ChannelCombo">
       <property name="geometry">
        <rect>
         <x>530</x>
         <y>30</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
      </widget>
      <zorder>DutyCycleGroup</zorder>
      <zorder>FrequencyGroup</zorder>
      <zorder>setButton</zorder>
      <zorder>PWMCheck</zorder>
      <zorder>ChannelCombo</zorder>
     </widget>
    </widget>
    <widget class="QWidget" name="TAB2">
     <attribute name="title">
      <string>PWM Peripherals</string>
     </attribute>
     <widget class="QCheckBox" name="PWMCheck_2">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>80</y>
        <width>163</width>
        <height>20</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Inverted PWM</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>19</x>
        <y>9</y>
        <width>191</width>
        <height>91</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QRadioButton" name="CTCRadioButton">
         <property name="text">
          <string>CTC/Normal Mode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="fastPWMRadio">
         <property name="text">
          <string>Fast PWM</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="PhaseCorrectPWM">
         <property name="text">
          <string>Phase Correct PWM</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="DutyCycleGroup_2">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>10</y>
        <width>241</width>
        <height>91</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
      <property name="title">
       <string>Duty Cycle (%)</string>
      </property>
      <widget class="QDoubleSpinBox" name="Dut_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>201</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="decimals">
        <number>0</number>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="value">
        <double>50.000000000000000</double>
       </property>
      </widget>
      <widget class="QSlider" name="DutyCycleSlider_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>191</width>
         <height>21</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBelow</enum>
       </property>
      </widget>
     </widget>
     <widget class="QComboBox" name="HardwareChannelsCombo">
      <property name="geometry">
       <rect>
        <x>700</x>
        <y>20</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="setButton_2">
      <property name="geometry">
       <rect>
        <x>700</x>
        <y>70</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QComboBox" name="PrescalerCombo">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>39</y>
        <width>163</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>10</y>
        <width>163</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Pre-scaler</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="LogGroupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>861</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Log</string>
    </property>
    <widget class="QTextEdit" name="logBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>841</width>
       <height>81</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="acceptRichText">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="uCfSpinBox">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>450</y>
      <width>291</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Microcontroller clock frequency (MHz)</string>
    </property>
    <widget class="QDoubleSpinBox" name="uCfSpin">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>221</width>
       <height>31</height>
      </rect>
     </property>
     <property name="decimals">
      <number>4</number>
     </property>
     <property name="minimum">
      <double>0.100000000000000</double>
     </property>
     <property name="maximum">
      <double>20.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>1.000000000000000</double>
     </property>
     <property name="value">
      <double>16.000000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>541</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Select Serial Port connected to the microcontroller</string>
    </property>
    <widget class="QComboBox" name="ChannelCombo_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>371</width>
       <height>30</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
    </widget>
    <widget class="QPushButton" name="checkCOMButton">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>30</y>
       <width>130</width>
       <height>30</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Check COM Ports</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="statusLabel">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>270</y>
      <width>391</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Frequency</string>
    </property>
   </widget>
   <zorder>tabWidget</zorder>
   <zorder>graphGroup</zorder>
   <zorder>LogGroupBox</zorder>
   <zorder>uCfSpinBox</zorder>
   <zorder>groupBox_3</zorder>
   <zorder>statusLabel</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>896</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionTutorial"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme=":/img/include/data-storage.png" resource="../../resources.qrc">
     <normaloff>:/images/internet.png</normaloff>:/images/internet.png</iconset>
   </property>
   <property name="text">
    <string>About..</string>
   </property>
   <property name="font">
    <font>
     <pointsize>9</pointsize>
    </font>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../../resources.qrc">
     <normaloff>:/images/floppy-disk.png</normaloff>:/images/floppy-disk.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="font">
    <font>
     <pointsize>9</pointsize>
    </font>
   </property>
  </action>
  <action name="actionTutorial">
   <property name="icon">
    <iconset resource="../../resources.qrc">
     <normaloff>:/images/round.png</normaloff>:/images/round.png</iconset>
   </property>
   <property name="text">
    <string>Tutorial</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="icon">
    <iconset resource="../../resources.qrc">
     <normaloff>:/images/data-storage.png</normaloff>:/images/data-storage.png</iconset>
   </property>
   <property name="text">
    <string>Load..</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>include/qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>FrequencySlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>frequencySpinUpdate(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FrequencySpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>frequencySliderUpdate(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Dut</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>dutySliderUpdate(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>396</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>525</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DutyCycleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>dutySpinUpdate(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>413</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>564</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PWMCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>flagAll(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>665</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>693</x>
     <y>520</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendToMicroSoftware()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>827</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>890</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkCOMButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>checkCOM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>407</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendToMicroHardware()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>769</x>
     <y>412</y>
    </hint>
    <hint type="destinationlabel">
     <x>895</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>tabBarClicked(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>HardwareTabGraph(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CTCRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>graphChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PrescalerCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>graphChangedproxy(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>470</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Dut_3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>dutySliderHardwareUpdate(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>509</x>
     <y>368</y>
    </hint>
    <hint type="destinationlabel">
     <x>551</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DutyCycleSlider_3</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>dutySpinUpdateHardware(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>469</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fastPWMRadio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>graphChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PhaseCorrectPWM</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>graphChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>408</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uCfSpin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>graphChangedproxy(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>518</y>
    </hint>
    <hint type="destinationlabel">
     <x>883</x>
     <y>559</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PWMCheck_2</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setFlagHardware(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>414</y>
    </hint>
    <hint type="destinationlabel">
     <x>289</x>
     <y>469</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>checkCOM()</slot>
  <slot>frequencySpinUpdate(int)</slot>
  <slot>frequencySliderUpdate(double)</slot>
  <slot>dutySliderUpdate(double)</slot>
  <slot>dutySpinUpdate(int)</slot>
  <slot>flagAll(bool)</slot>
  <slot>sendToMicroSoftware()</slot>
  <slot>sendToMicroHardware()</slot>
  <slot>HardwareTabGraph(int)</slot>
  <slot>graphChanged()</slot>
  <slot>graphChangedproxy(int)</slot>
  <slot>dutySliderHardwareUpdate(double)</slot>
  <slot>dutySpinUpdateHardware(int)</slot>
  <slot>setFlagHardware(bool)</slot>
  <slot>graphChangedproxy(double)</slot>
 </slots>
</ui>
